### PR TITLE
Version 7.2.0

### PR DESCRIPTION
**Pull request by mwislek**
The following changes were made in this update:
- All dependencies were updated to their latest minor versions
- The following major dependency updates were performed:
  - `commander`
  - `fs-extra`. This update fixes an issue where the `Layout.js` filename was being changed after a copy. This library is highly dependent on the Node version and was very out-of-date with respect to the current versions of Node, which  corrected this issue.
  - `glob`
  - `pluralize`
 - The following issues were resolved:
   - [Incorrect Layout Filename](https://github.com/LaunchPadLab/lp-subsection-generator/issues/28)
   - [Make reducer's reducerKey and slice match given subsection name ](https://github.com/LaunchPadLab/lp-subsection-generator/issues/19)
 - The "Show" views were updated to use the `useParams()` hook to obtain the resource `id` rather then the HOC previously used
 - Line spacing was changed to be more consistent across all files.

Note that both the client template and the Ionic template should be updated to use this newest version.

### CODE REVIEW

File: generate.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ function generate (subSectionName, destination, options) {
   allFiles.forEach(file => {
     const template = fs.readFileSync(file, 'utf8')
     const result = template
+      .replace(/%subsection%/g, subSectionName)
```
Comment (mwislek) [inline generate.js:29]: Part of resolving this issue: https://github.com/LaunchPadLab/lp-subsection-generator/issues/19

File: ionic-template/reducer.js [hunk 2]
```diff
@@ -4,19 +4,15 @@ import { handleActions } from 'redux-actions'
 // import * as actions from './actions'
 // import * as apiActions from 'api-actions'
 
-const reducerKey = '%subSection%'
-// const slice = 'root.%subSection%'
-
+const reducerKey = '%subsection%'
```
Comment (mwislek) [inline ionic-template/reducer.js:7]: Same change for Ionic.

File: ionic-template/views/SubSectionShow.js [hunk 3]
```diff
@@ -11,16 +11,14 @@ import { IonItem, IonLabel, IonList, IonListHeader } from '@ionic/react'
 // import * as Types from 'types'
 
 const propTypes = {
-  // id: PropTypes.number.isRequired,
```
Comment (mwislek) [inline ionic-template/views/SubSectionShow.js:14]: No longer needed.

File: ionic-template/views/SubSections.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { IonItem, IonLabel, IonList, IonListHeader } from '@ionic/react'
 // import * as apiActions from 'api-actions'
 
 const propTypes = {
-  // %subSections%: PropTypes.arrayOf(Types.%subSection%),
+  // %subSections%: PropTypes.arrayOf(Types.%SubSection%),
```
Comment (mwislek) [inline ionic-template/views/SubSections.js:13]: Types should follow the JS convention of beginning with a capital letter (class format).

File: template/reducer.js [hunk 5]
```diff
@@ -4,19 +4,15 @@ import { handleActions } from 'redux-actions'
 // import * as actions from './actions'
 // import * as apiActions from 'api-actions'
 
-const reducerKey = '%subSection%'
-// const slice = 'root.%subSection%'
-
+const reducerKey = '%subsection%'
```
Comment (mwislek) [inline template/reducer.js:7]: Related to correction of reducer key issue.

File: template/views/SubSectionShow.js [hunk 6]
```diff
@@ -2,43 +2,43 @@ import React /*, { useEffect } */ from 'react'
 // import PropTypes from 'prop-types'
 import { compose } from 'redux'
 import { connect } from 'react-redux'
+import { useParams } from 'react-router-dom'
```
Comment (mwislek) [inline template/views/SubSectionShow.js:5]: Use a hook rather than the HOC.

File: MIGRATION.md [hunk 7]
```diff
@@ -1,5 +1,14 @@
 # Migration Guide
 
+## Migration to Version ^8.0.0
+Version 8 introduces major updates of several libraries:
+- `commander`
+- `fs-extra`. This update fixes an issue where the `Layout.js` filename was being changed after a copy. This library is highly dependent on the Node version and was very out-of-date with respect to the current versions of Node, which  corrected this issue.
+- `glob`
+- `pluralize`
+
+There should should be no manual migration required by the user due to these changes.
```
Comment (chawes13) [inline MIGRATION.md:10]: Can we lead with this?

File: ionic-template/reducer.js [hunk 8]
```diff
@@ -6,17 +6,13 @@ import { handleActions } from 'redux-actions'
 
 const reducerKey = '%subSection%'
 // const slice = 'root.%subSection%'
-
```
[the remark this answers is not in the document]
Comment (chawes13) [reply ionic-template/reducer.js:9]: This is so niche. I love it!

File: MIGRATION.md (hunk 7 above)
Comment (chawes13) [reply MIGRATION.md:10]: If there is no change needed on the user's end, is it really a breaking change? 🤔

File: ionic-template/views/SubSectionShow.js (hunk 3 above)
Comment (chawes13) [reply ionic-template/views/SubSectionShow.js:14]: Nice catch

File: ionic-template/views/SubSections.js (hunk 4 above)
Comment (chawes13) [reply ionic-template/views/SubSections.js:13]: Hmm, should they? Can you send me a reference? I've never followed this convention.

File: package.json [hunk 9]
```diff
@@ -36,6 +36,6 @@
     "redux-actions": "^2.6.5"
   },
   "optionalDependencies": {
-    "@ionic/react": "^5.0.0"
+    "@ionic/react": "^6.4.0"
```
Comment (chawes13) [inline package.json:39]: Do we have to drop the support for 5?

File: MIGRATION.md (hunk 7 above)
Comment (mwislek) [reply MIGRATION.md:10]: Maybe not. None of the major upgrades required any changes to the code. And, in one case (the strange problem with the `Layout.js` file's case becoming `layout.js`), the issue went away with no other change made by me. I'll revert to a minor update.

File: ionic-template/views/SubSections.js (hunk 4 above)
Comment (mwislek) [reply ionic-template/views/SubSections.js:13]: These are a bit strange. They are constant definitions but they are NOT constants. They are objects that represent a type, used for type checking. See: https://google.github.io/styleguide/jsguide.html#naming-rules-by-identifier-type. In my mind, these are similar to classes. The current convention of naming these starting with lower-case didn't really highlight their "typeness" to me. Rather, this convention made them look like locally defined constant variables, which they are not.

File: package.json (hunk 9 above)
Comment (mwislek) [reply package.json:39]: I guess we could do `@ionic/react >= 5.0.0` or something. But this version of the this library will be integrated immediately into the ionic-client-template, which already was upgraded with @ionic/react version V ^6.0.0.

File: ionic-template/views/SubSections.js (hunk 4 above)
Comment (chawes13) [reply ionic-template/views/SubSections.js:13]: We should paint the shed blue! That link is helpful. Let's do it.

File: package.json (hunk 9 above)
Comment (chawes13) [reply package.json:39]: Yes, but there are apps built with the older version of the ionic-client-template in the wild, eh? If a user had an app with ^5 and looked at this change, they wouldn't think it was safe to upgrade. In fact, if we don't offer backwards compatibility for ionic, I'd lean towards making it a major upgrade.

But what's the benefit of that? Seems like a hassle to me.

Comment (mwislek) [reply package.json:39]: Updated to > 5. I also made a few more updates that were changed as part of the client-template update.

File: package.json [hunk 10]
```diff
@@ -18,24 +18,24 @@
...
     "react": "^16.8.0 || ^17.0.0",
-    "react-redux": "^7.2.8",
+    "react-redux": "^7.2.8 || ^8.0.0",
     "react-router-dom": "^4.0.0 || ^5.0.0",
     "redux": "^4.1.2",
     "redux-actions": "^2.6.5"
   },
   "optionalDependencies": {
-    "@ionic/react": "^5.0.0"
+    "@ionic/react": ">= 5"
```
Comment (mwislek) [inline package.json:39]: Updated per previous comment.

File: package.json [hunk 11]
```diff
@@ -18,24 +18,24 @@
...
   },
   "devDependencies": {
     "@launchpadlab/eslint-config": "^2.1.0",
-    "eslint": "^4.11.0"
+    "eslint": "^7.32.0"
   },
   "peerDependencies": {
-    "react": "^16.8.0 || ^17.0.0",
-    "react-redux": "^7.2.8",
+    "react": ">= 16",
```
Comment (mwislek) [inline package.json:32]: To accommodate the Ionic client template (which **_is_** on React 18)

File: ionic-template/Routes.js [hunk 12]
```diff
@@ -13,7 +13,7 @@ function Routes () {
   return (
     <Layout>
       <IonRouterOutlet>
-        <Route exact path={`${path}/`}>
+        <Route exact path={path}>
```
Comment (mwislek) [inline ionic-template/Routes.js:16]: Requested by Ji. This is now aligned with the client template.

**Review by chawes13 — APPROVED** [review]
Nice! Please update the PR title before merging 🙂

**Pull request merged**